bench_arrayproxy_slicing.py fails

On both Linux and MacOS, downloading the latest version of nibabel generates failures with the command:
  time pytest -c ./pytest.benchmark.ini ./bench_arrayproxy_slicing.py

```
>time pytest -c ./pytest.benchmark.ini ./bench_arrayproxy_slicing.py
============================================================= test session starts ==============================================================
platform darwin -- Python 3.7.4, pytest-6.0.1, py-1.9.0, pluggy-0.13.1
rootdir: /Users/chris/Downloads/nibabel/nibabel/benchmarks, configfile: pytest.benchmark.ini
collected 1 item                                                                                                                               

bench_arrayproxy_slicing.py 
ArrayProxy gzip slicing for git revision 591a1bd
-------------------------------------------------
Generating test data... (381 MB)
Running test 1 of 15 (gzip)...
Running test 2 of 15 (gzip)...
Running test 3 of 15 (gzip)...
Running test 4 of 15 (gzip)...
Running test 5 of 15 (gzip)...
Running test 6 of 15 (gzip)...
Running test 7 of 15 (gzip)...
F

=================================================================== FAILURES ===================================================================
___________________________________________________________ bench_arrayproxy_slicing ___________________________________________________________

    def bench_arrayproxy_slicing():
    
        print_git_title('\nArrayProxy gzip slicing')
    
        # each test is a tuple containing
        # (HAVE_INDEXED_GZIP, keep_file_open, sliceobj)
        tests = list(it.product(HAVE_IGZIP, KEEP_OPENS, SLICEOBJS))
    
        # remove tests where HAVE_INDEXED_GZIP is True and keep_file_open is False,
        # because if keep_file_open is False, HAVE_INDEXED_GZIP has no effect
        tests = [t for t in tests if not (t[0] and not t[1])]
    
        testfile = 'testfile.nii'
        testfilegz = 'test.nii.gz'
    
        def get_test_label(test):
            have_igzip = test[0]
            keep_open = test[1]
    
            if not (have_igzip and keep_open):
                return 'gzip'
            else:
                return 'indexed_gzip'
    
        def fix_sliceobj(sliceobj):
            new_sliceobj = []
            for i, s in enumerate(sliceobj):
                if s == ':':
                    new_sliceobj.append(slice(None))
                elif s == '?':
                    new_sliceobj.append(np.random.randint(0, SHAPE[i]))
                else:
                    new_sliceobj.append(int(s * SHAPE[i]))
            return tuple(new_sliceobj)
    
        def fmt_sliceobj(sliceobj):
            slcstr = []
            for i, s in enumerate(sliceobj):
                if s in ':?':
                    slcstr.append(s)
                else:
                    slcstr.append(str(int(s * SHAPE[i])))
            return f"[{', '.join(slcstr)}]"
    
        with InTemporaryDirectory():
    
            print(f'Generating test data... ({int(round(np.prod(SHAPE) * 4 / 1048576.0))} MB)')
    
            data = np.array(np.random.random(SHAPE), dtype=np.float32)
    
            # zero out 10% of voxels so gzip has something to compress
            mask = np.random.random(SHAPE[:3]) > 0.1
            if len(SHAPE) > 3:
                data[mask, :] = 0
            else:
                data[mask] = 0
    
            # save uncompressed and compressed versions of the image
            img = nib.nifti1.Nifti1Image(data, np.eye(4))
            nib.save(img, testfilegz)
            nib.save(img, testfile)
    
            # each result is a tuple containing
            # (label, keep_open, sliceobj, testtime, basetime, testmem, basemem)
            #
            # where "basetime" is the time taken to load and slice a memmapped
            # (uncompressed)image, and "basemem" is memory usage for the same
            results = []
    
            # We use the same random seed for each slice object,
            seeds = [np.random.randint(0, 2 ** 32) for s in SLICEOBJS]
    
            for ti, test in enumerate(tests):
    
                label = get_test_label(test)
                have_igzip, keep_open, sliceobj = test
                seed = seeds[SLICEOBJS.index(sliceobj)]
    
                print(f'Running test {ti + 1} of {len(tests)} ({label})...')
    
                # load uncompressed and compressed versions of the image
>               img = nib.load(testfile, keep_file_open=keep_open)

bench_arrayproxy_slicing.py:138: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../loadsave.py:52: in load
    img = image_klass.from_filename(filename, **kwargs)
../dataobj_images.py:489: in from_filename
    keep_file_open=keep_file_open)
../analyze.py:976: in from_file_map
    keep_file_open=keep_file_open)
../arrayproxy.py:165: in __init__
    self._should_keep_file_open(file_like, keep_file_open)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <nibabel.arrayproxy.ArrayProxy object at 0x115ceee50>, file_like = 'testfile.nii', keep_file_open = 'auto'

    def _should_keep_file_open(self, file_like, keep_file_open):
        """Called by ``__init__``.
    
        This method determines how to manage ``ImageOpener`` instances,
        and the underlying file handles - the behaviour depends on:
    
         - whether ``file_like`` is an an open file handle, or a path to a
           ``'.gz'`` file, or a path to a non-gzip file.
         - whether ``indexed_gzip`` is present (see
           :attr:`.openers.HAVE_INDEXED_GZIP`).
    
        An ``ArrayProxy`` object uses two internal flags to manage
        ``ImageOpener`` instances and underlying file handles.
    
          - The ``_persist_opener`` flag controls whether a single
            ``ImageOpener`` should be created and used for the lifetime of
            this ``ArrayProxy``, or whether separate ``ImageOpener`` instances
            should be created on each file access.
    
          - The ``_keep_file_open`` flag controls qwhether the underlying file
            handle should be kept open for the lifetime of this
            ``ArrayProxy``, or whether the file handle should be (re-)opened
            and closed on each file access.
    
        The internal ``_keep_file_open`` flag is only relevant if
        ``file_like`` is a ``'.gz'`` file, and the ``indexed_gzip`` library is
        present.
    
        This method returns the values to be used for the internal
        ``_persist_opener`` and ``_keep_file_open`` flags; these values are
        derived according to the following rules:
    
        1. If ``file_like`` is a file(-like) object, both flags are set to
        ``False``.
    
        2. If ``keep_file_open`` (as passed to :meth:``__init__``) is
           ``True``, both internal flags are set to ``True``.
    
        3. If ``keep_file_open`` is ``False``, but ``file_like`` is not a path
           to a ``.gz`` file or ``indexed_gzip`` is not present, both flags
           are set to ``False``.
    
        4. If ``keep_file_open`` is ``False``, ``file_like`` is a path to a
           ``.gz`` file, and ``indexed_gzip`` is present, ``_persist_opener``
           is set to ``True``, and ``_keep_file_open`` is set to ``False``.
           In this case, file handle management is delegated to the
           ``indexed_gzip`` library.
    
        .. deprecated:: 2.4.1
            ``keep_file_open='auto'`` is redundant with `False` and has
            been deprecated. It raises an error as of nibabel 3.0.
    
        Parameters
        ----------
    
        file_like : object
            File-like object or filename, as passed to ``__init__``.
        keep_file_open : { True, False }
            Flag as passed to ``__init__``.
    
        Returns
        -------
    
        A tuple containing:
          - ``keep_file_open`` flag to control persistence of file handles
          - ``persist_opener`` flag to control persistence of ``ImageOpener``
            objects.
        """
        if keep_file_open is None:
            keep_file_open = KEEP_FILE_OPEN_DEFAULT
            if keep_file_open not in (True, False):
                raise ValueError("nibabel.arrayproxy.KEEP_FILE_OPEN_DEFAULT "
                                 f"must be boolean. Found: {keep_file_open}")
        elif keep_file_open not in (True, False):
>           raise ValueError('keep_file_open must be one of {None, True, False}')
E           ValueError: keep_file_open must be one of {None, True, False}

../arrayproxy.py:261: ValueError
=========================================================== short test summary info ============================================================
FAILED bench_arrayproxy_slicing.py::bench_arrayproxy_slicing - ValueError: keep_file_open must be one of {None, True, False}
======================================================== 1 failed in 933.90s (0:15:33) =========================================================

```

--- a/Changelog
+++ b/Changelog
@@ -25,8 +25,20 @@ Eric Larson (EL), Demian Wassermann, Stephan Gerhard and Ross Markello (RM).
 
 References like "pr/298" refer to github pull request numbers.
 
+3.2.1 (Saturday 28 November 2020)
+=================================
+
+Bug fix release in the 3.2.x series.
+
+Maintenance
+-----------
+* Drop references to builtin types in Numpy namespace like ``np.float``
+  (pr/964) (EL, reviewed by CM)
+* Ensure compatibility with Python 3.9 (pr/963) (CM)
+
+
 3.2.0 (Tuesday 20 October 2020)
-==============================
+===============================
 
 New feature release in the 3.2.x series.
 

--- a/nibabel/benchmarks/bench_arrayproxy_slicing.py
+++ b/nibabel/benchmarks/bench_arrayproxy_slicing.py
@@ -46,7 +46,7 @@ SLICEOBJS = [
     ('?', '?', '?', ':'),
 ]
 
-KEEP_OPENS = [False, True, 'auto']
+KEEP_OPENS = [False, True]
 
 if HAVE_INDEXED_GZIP:
     HAVE_IGZIP = [False, True]

--- a/setup.cfg
+++ b/setup.cfg
@@ -16,6 +16,7 @@ classifiers =
     Programming Language :: Python :: 3.6
     Programming Language :: Python :: 3.7
     Programming Language :: Python :: 3.8
+    Programming Language :: Python :: 3.9
     Topic :: Scientific/Engineering
 license = MIT License
 description = Access a multitude of neuroimaging data formats

